### DISCUSSION TITLE
Creating a custom watch command

### DISCUSSION THREAD

**Discussion by mauriceoc** [nrwl/nx#17778] [General]
I have a bunch of JSON files in my project, taking one as an example: `file1.json`.

As part of the build I would like to run a transformation on this and output the transformed data into a new json file:

`file1-transformed.js`

This transfomed file is imported by my app.

I have this working using an executor called `run-transform`. When the app is built then `file1-transformed.js` is created:

```
nx run foobar:run-transform
```

However if I run the `dev-server` then `file1.json` is not counted in the watch process. If I modify `file1.json` the transformation is not executed and the change is not reflected in my locally running browser.

What is the best way to configure NX in order to detect changes to `file1.json` and run the build / update the running application when it is modified?

I would prefer not to run a separate watch process for this if I can avoid it.

**Comment by jbjhjm** [comment]
I'd be interested in that as well. AFAIK the current executors cannot handle this.

To be included in any watching, it seems one has to manipulate the Project Graph.
This might be possible through a Project Graph plugin, documents have a rough guide on this.
With this implemented correctly, a change to your source file should re-trigger compilation when watching.

This does however not include your run-transform executor I think.
I'm really not sure if there is any method in place to re-run/watch custom executors.
Right now I'm digging through source code of the js/node executor, analyzing how to set up watching/executing manually.
